frontend/src/metabase-lib/v1/parameters/utils/targets.ts

```diff
@@ -60,11 +60,16 @@ export function getParameterTargetField(
   if (isConcreteFieldReference(fieldRef)) {
     const fieldId = fieldRef[1];
     const resultMetadata = question.getResultMetadata();
-    const fieldMetadata = resultMetadata.find(field => field.id === fieldId);
-    return (
-      metadata.field(fieldId, fieldMetadata?.table_id) ??
-      metadata.field(fieldId)
+    const fieldMetadata = resultMetadata.find(
+      field => field.id === fieldId || field.name === fieldId,
```

Comment:
This matches MLv1 behavior https://github.com/metabase/metabase/blob/17f960c0d03cfb6f3e0835b780a6929bc0035060/frontend/src/metabase-lib/v1/Dimension.ts#L794